include/ringmesh/mesh/cartesian_grid.h

```diff
@@ -281,11 +281,160 @@ namespace RINGMesh
          * @return the attribute manager containing the grid point attributes
          * values.
          */
-        GEO::AttributesManager& attributes_manager() const
+        GEO::AttributesManager& attributes_manager()
         {
             return attributes_manager_;
         }
 
+        /*!
+         * @return the attribute manager containing the grid point attributes
+         * values.
+         */
+        const GEO::AttributesManager& attributes_manager() const
+        {
+            return attributes_manager_;
+        }
+
+        void add_index_t_attribute( const std::string& attribute_name,
+            const std::vector< index_t >& values )
+        {
+            if( values.size() != nb_total_cells_ )
+            {
+                throw RINGMeshException( "CartesianGrid",
+                    "The attribute you're trying to add doesn't have the same"
+                    " number of values than the grid." );
+            }
+            attributes_manager_.bind_attribute_store( attribute_name,
+                GEO::AttributeStore::
+                    create_attribute_store_by_element_type_name(
+                        "index_t", 1 ) );
+            GEO::Attribute< index_t > attribute{ attributes_manager_,
+                attribute_name };
+            for( auto i : range( nb_total_cells_ ) )
+            {
+                attribute[i] = values[i];
+            }
+        }
+
+        void add_float_attribute( const std::string& attribute_name,
+            const std::vector< float >& values )
+        {
+            if( values.size() != nb_total_cells_ )
```

Comment:
Is this statement possible? If it should never happen => assert. In which cases this case can happen and can we avoid that?

Reply:
This statement is possible if the user gives a vector of attribute values of the wrong size, I will put an assert since it should indeed never happen (or rather, it must never happen)

Reply:
When you say user, do you mean a developer who uses this function? If yes, we do have to prevent this case. The developer has to make sure this statement never happens. So I do think you need an assert. For me a throw is only when a real user (= not a developer but someone using a binary) loads for instance a grid from a file and somehow the file has a problem (nb cells != nb values). You can argue that you check such a potential issue here but this function and this class look very low-level to me. Such a check on a i/o should be high level (e.g. in the main). Please clarify what is an user here and in which cases this statement can be true.

Reply:
Well this is needed for a developper, but I think also for a real user : if he wants for any reason to add a new attribute to a grid he is using, there must be something preventing him from adding an attribute with the wrong number of values.

Reply:
for a user, do you have a main to add a value? if no, that means that you have no user, else that means you have to put the throw in a high level function (e.g., the main). I do think that such a throw should not be in a very low-level function.